src/components/organisms/PrivateChatModal/PrivateChatModal.tsx

```diff
@@ -59,6 +59,8 @@ const PrivateChatModal: React.FunctionComponent = () => {
     }, {});
 
   const onClickOnSender = (sender: WithId<User>) => {
+    if (!privateChats) return;
+
     const chatsToUpdate = privateChats.filter(
       (chat) => !chat.isRead && chat.from === sender.id
     );
```

Comment:
[DRY] This looks like the same code as used in `src/components/molecules/ChatsList/ChatsList.tsx`. We should extract it to a common place.

Reply:
Agreed. Thinking on incremental v rewrite. The sheer quantity of these types of nits makes me lean in favour of a rebuild of key components in compliance with our new, higher standards.

Reply:
To me, that still fits within what I would call an 'incremental refactor' rather than a 'rewrite' (which to me implies starting from scratch entirely). May just be a difference in semantics though.

Similar to how I 'ripped apart' the theme camp template as part of my performance refactors, that is what I think we should be doing to the other core components we need up to scratch.

Reply:
If you could create a shared list of the components you think fall under this 'need rewrites' (on trello/similar?) then that would help with prioritising + assessing effort required.